certz/certz.proto

```diff
@@ -636,8 +641,84 @@ message CertificateSigningRequest {
 // the specified parameters.
 message GenerateCSRRequest {
   CSRParams params = 1;
+
+  // Indicates measurements that are prerequisites for certificate
+  // provisioning.
+  repeated ReferenceIntegritySpec integrity = 2;
 }
+
 // A message sent by the target as response to the GenerateCSRRequest.
 message GenerateCSRResponse {
   CertificateSigningRequest certificate_signing_request = 1;
+
+  // Reference integrity measurements as specified in the request.
+  repeated ReferenceIntegrityData integrity = 2;
+}
+
+message GetIntegrityManifestRequest {}
+
+message GetIntegrityManifestResponse {
+  bytes manifest = 1;
+}
+
+// See the TCG Reference Integrity Manifest (RIM) Information Model.
+message ReferenceIntegritySpec {
+  // Identifies the Root of Trust component.
```

Comment:
what are the list of components we consider valid?
should this be a string or a enum

Reply:
As discussed offline, this should remain a string.